nmdc_server/database.py

```diff
@@ -128,8 +125,14 @@
     END LOOP;
 END;
 $$ LANGUAGE plpgsql;
-"""
-    ),
+    """
+)
+
+# define functions used for comparison between arbitrary types
+listen(
+    metadata,
+    "before_create",
+    update_nmdc_functions_sql,
```

Comment:
So you are leaving in this listen even if we don't know if it's ever triggered. Sounds reasonable and a safe choice just to make sure we aren't breaking existing functionality. At worst we may now be sometimes updating functions twice but it's a very short and reentrant operation so that's ok.

Reply:
Yes, that sums up my reasoning as well.